infra/lib/ci-cluster.ts

```diff
@@ -1,58 +1,80 @@
 import { Construct } from "constructs";
-import { aws_eks as eks, aws_ec2 as ec2, aws_iam as iam } from "aws-cdk-lib";
+import { aws_eks as eks, aws_ec2 as ec2, Stack } from "aws-cdk-lib";
+import * as blueprints from "@aws-quickstart/eks-blueprints";
 import * as cdk8s from "cdk8s";
-import { policies as ALBPolicies } from "./policies/aws-load-balancer-controller-policy";
 import {
   ProwGitHubSecretsChart,
   ProwGitHubSecretsChartProps,
 } from "./charts/prow-secrets";
 import {
-  EXTERNAL_DNS_NAMESPACE,
   FLUX_NAMESPACE,
   PROW_JOB_NAMESPACE,
   PROW_NAMESPACE,
 } from "./test-ci-stack";
+import {
+  GlobalResources,
+  ImportHostedZoneProvider,
+} from "@aws-quickstart/eks-blueprints";
 
-export type CIClusterCompileTimeProps = ProwGitHubSecretsChartProps;
+export type CIClusterCompileTimeProps = ProwGitHubSecretsChartProps & {
+  hostedZoneId: string;
+};
 
 export type CIClusterRuntimeProps = {};
 
 export type CIClusterProps = CIClusterCompileTimeProps & CIClusterRuntimeProps;
 
 export class CICluster extends Construct {
   readonly testCluster: eks.Cluster;
-  readonly testNodegroup: eks.Nodegroup;
 
   readonly namespaceManifests: eks.KubernetesManifest[];
 
   constructor(scope: Construct, id: string, props: CIClusterProps) {
     super(scope, id);
 
-    this.testCluster = new eks.Cluster(scope, "TestInfraCluster", {
-      version: eks.KubernetesVersion.V1_24,
-      defaultCapacity: 0,
-    });
-    this.testNodegroup = this.testCluster.addNodegroupCapacity(
-      "TestInfraNodegroup",
-      {
-        instanceTypes: [
-          ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.InstanceSize.XLARGE8),
-        ],
-        minSize: 2,
-        diskSize: 150,
-      }
+    const clusterVersion = eks.KubernetesVersion.V1_23;
+
+    const mngProps: blueprints.MngClusterProviderProps = {
+      minSize: 2,
+      maxSize: 8,
+      desiredSize: 2,
+      diskSize: 150,
+      version: clusterVersion,
+      instanceTypes: [
+        ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.InstanceSize.XLARGE8),
+      ],
+      amiType: eks.NodegroupAmiType.AL2_X86_64,
+      nodeGroupCapacityType: eks.CapacityType.ON_DEMAND,
+    };
+
+    const blueprintStack = blueprints.EksBlueprint.builder()
+      .account(Stack.of(this).account)
+      .region(Stack.of(this).region)
+      .version(clusterVersion)
+      .clusterProvider(new blueprints.MngClusterProvider(mngProps))
+      .resourceProvider(
+        GlobalResources.HostedZone,
+        new ImportHostedZoneProvider(props.hostedZoneId)
+      )
+      .addOns(
+        new blueprints.addons.VpcCniAddOn(),
+        new blueprints.addons.KarpenterAddOn(),
+        new blueprints.addons.AwsLoadBalancerControllerAddOn(),
+        new blueprints.addons.EbsCsiDriverAddOn(),
```

Comment:
I think it's kinda crazy that the EBS CSI driver is no longer installed by default in EKS clusters. In fact, that is pointed out in [this article](https://medium.com/@elliotgraebert/comparing-the-top-eight-managed-kubernetes-providers-2ae39662391b) specifically as causing major headaches compared to GKE and AKS...